saos-webapp/src/test/java/pl/edu/icm/saos/webapp/analysis/AnalysisControllerTest.java

```diff
@@ -0,0 +1,1211 @@
+        assertAggregatedChartSeriesDouble(0, result, 1, avgSeries0);
+        
+        double avgSeries1 = series1Values.stream().collect(Collectors.averagingDouble(x -> x));
+        assertAggregatedChartSeriesDouble(1, result, 3, avgSeries1);
+    }
+    
+    @Test
+    public void generate_ForCommonCourts() throws Exception {
+        // execute
+        ResultActions result = mockMvc.perform(get(URL_BASE + "/generate")
+                .param("seriesFilters[0].phrase", "phrase1")
+                .param("seriesFilters[1].phrase", "phrase2")
+                .param("globalFilter.judgmentDateRange.startMonth", "1")
+                .param("globalFilter.judgmentDateRange.startYear", "2000")
+                .param("globalFilter.judgmentDateRange.endMonth", "1")
+                .param("globalFilter.judgmentDateRange.endYear", "2001")
+                .param("globalFilter.courtCriteria.courtType", "COMMON")
+                .param("ysettings.valueType", "NUMBER")
+                .accept(MediaType.APPLICATION_JSON));
+        
+        
+        // assert
+        
+        assertMainChartSeries(0, result, 4, 0, 0, 6, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+        assertMainChartSeries(1, result, 5, 0, 0, 4, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+        
+        assertMainChartXticksFor2000Year(result);
+        
+        
+        assertAggregatedChartSeries(0, result, 1, 10);
+        assertAggregatedChartSeries(1, result, 3, 9);
+        
+        
+        assertCcCourtChartSeries(0, result, 8, 2);
+        assertCcCourtChartSeries(1, result, 8, 1);
+        
+        assertCcCourtChartXticks(result, appealCourt1, appealCourt2);
+    }
+    
+    @Test
+    public void generate_ForCommonCourt() throws Exception {
+        // execute
+        ResultActions result = mockMvc.perform(get(URL_BASE + "/generate")
+                .param("seriesFilters[0].phrase", "phrase1")
+                .param("seriesFilters[1].phrase", "phrase2")
+                .param("globalFilter.judgmentDateRange.startMonth", "1")
+                .param("globalFilter.judgmentDateRange.startYear", "2000")
+                .param("globalFilter.judgmentDateRange.endMonth", "1")
+                .param("globalFilter.judgmentDateRange.endYear", "2001")
+                .param("globalFilter.courtCriteria.courtType", "COMMON")
+                .param("globalFilter.courtCriteria.ccCourtId", String.valueOf(appealCourt1.getId()))
+                .param("ysettings.valueType", "NUMBER")
+                .accept(MediaType.APPLICATION_JSON));
+        
+        
+        // assert
+        
+        assertMainChartSeries(0, result, 3, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+        assertMainChartSeries(1, result, 5, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+        
+        assertMainChartXticksFor2000Year(result);
+        
+        assertAggregatedChartSeries(0, result, 1, 5);
+        assertAggregatedChartSeries(1, result, 3, 5);
+    }
```

Comment:
You probably wanted to check here the ccCourtChart with assertCcCourtChartSeries/Xticks